apa102/apa102.go

```diff
@@ -0,0 +1,75 @@
+// Package apa102 implements a driver for the APA102 SPI LED.
+//
+// Datasheet: https://cdn-shop.adafruit.com/product-files/2343/APA102C.pdf
+package apa102
+
+import (
+	"image/color"
+	"machine"
+)
+
+const (
+	// BGR aka "Blue Green Red" is the current APA102 LED color order.
+	BGR = iota
+
+	// BRG aka "Blue Red Green" is the typical APA102 color order from 2015-2017.
+	BRG
+
+	// GRB aka "Green Red Blue" is the typical APA102 color order from pre-2015.
+	GRB
+)
+
+// Device wraps APA102 SPI LEDs.
+type Device struct {
+	bus   machine.SPI
+	tx    []byte
+	count int
+	Order int
+}
+
+// New returns a new APA102 driver. Pass in a fully configured SPI bus, and the count of
+// APA102 LEDs that are connected together.
+func New(b machine.SPI, count int) Device {
+	t := make([]byte, count*4)
```

Comment:
I would really like to see this allocation gone, as RAM is scarce and allocations are expensive. I think it can be done at the cost of a bit of code duplication. Or is there a reason why a buffer has to be allocated first?

Reply:
OK I made change to accommodate this idea. This will certainly help for the case that larger numbers of LEDs are being used.